x/lockup/keeper/lock_test.go

```diff
@@ -333,64 +332,182 @@ func (suite *KeeperTestSuite) TestCreateLock() {
+		expectAddTokensToLockSuccess bool
+		expectedError                bool
+	}{
+		{
+			name:                         "normal add tokens to lock case",
+			tokenToAdd:                   initialLockCoin,
+			duration:                     time.Second,
+			lockingAddress:               addr1,
+			expectAddTokensToLockSuccess: true,
+		},
+		{
+			name:           "not the owner of the lock",
+			tokenToAdd:     initialLockCoin,
+			duration:       time.Second,
+			lockingAddress: addr2,
+			expectedError:  true,
+		},
+		{
+			name:           "lock with matching duration not existing",
+			tokenToAdd:     initialLockCoin,
+			duration:       time.Second * 2,
+			lockingAddress: addr1,
+			expectedError:  true,
+		},
+		{
+			name:           "lock invalid tokens",
+			tokenToAdd:     sdk.NewCoin("unknown", sdk.NewInt(10)),
+			duration:       time.Second,
+			lockingAddress: addr1,
+			expectedError:  true,
+		},
+		{
+			name:           "token to add exceeds balance",
+			tokenToAdd:     sdk.NewCoin("stake", sdk.NewInt(20)),
+			duration:       time.Second,
+			lockingAddress: addr1,
+			expectedError:  true,
+		},
+	}
 
-	// add more tokens to lock
-	addCoins := sdk.NewInt64Coin("stake", 10)
-	suite.FundAcc(addr1, sdk.Coins{addCoins})
-	_, err = suite.App.LockupKeeper.AddTokensToLockByID(suite.Ctx, locks[0].ID, addr1, addCoins)
-	suite.Require().NoError(err)
+	for _, tc := range testCases {
+		suite.SetupTest()
+		// lock with balance
+		suite.FundAcc(addr1, sdk.Coins{initialLockCoin})
+		originalLock, err := suite.App.LockupKeeper.CreateLock(suite.Ctx, addr1, sdk.Coins{initialLockCoin}, time.Second)
+		suite.Require().NoError(err)
 
-	// check locks after adding tokens to lock
-	locks, err = suite.App.LockupKeeper.GetPeriodLocks(suite.Ctx)
-	suite.Require().NoError(err)
-	suite.Require().Len(locks, 1)
-	suite.Require().Equal(locks[0].Coins, coins.Add(sdk.Coins{addCoins}...))
+		suite.FundAcc(addr1, sdk.Coins{initialLockCoin})
+		balanceBeforeLock := suite.App.BankKeeper.GetAllBalances(suite.Ctx, tc.lockingAddress)
 
-	// check accumulation store is correctly updated
-	accum = suite.App.LockupKeeper.GetPeriodLocksAccumulation(suite.Ctx, types.QueryCondition{
-		LockQueryType: types.ByDuration,
-		Denom:         "stake",
-		Duration:      time.Second,
-	})
-	suite.Require().Equal(accum.String(), "20")
+		lockID, err := suite.App.LockupKeeper.AddToExistingLock(suite.Ctx, tc.lockingAddress, tc.tokenToAdd, tc.duration)
+		if tc.expectedError {
+			suite.Require().Error(err)
+		} else {
+			suite.Require().NoError(err)
+		}
 
-	// try to add tokens to unavailable lock
-	cacheCtx, _ := suite.Ctx.CacheContext()
-	err = simapp.FundAccount(suite.App.BankKeeper, cacheCtx, addr1, sdk.Coins{addCoins})
-	suite.Require().NoError(err)
-	// curBalance := suite.App.BankKeeper.GetAllBalances(cacheCtx, addr1)
-	_, err = suite.App.LockupKeeper.AddTokensToLockByID(cacheCtx, 1111, addr1, addCoins)
-	suite.Require().Error(err)
+		if tc.expectAddTokensToLockSuccess {
```

Comment:
What is the difference between `expectError` and `expectAddTokensToLockSuccess`? Can we use only `expectError`?

Reply:
good catch, the original intent was to create a separate case for lock failing and adding to lock failing, but these can be grouped to one. Changed to removing the field!